Security Trainees

Moonyman opened the pull request:
All other departments have trainees, security is no worse. Opened in loving memory of those poor crewmen who are willing to spend half of the round sitting in front of the XO's desk and by security to get their security trainee status only to get forgotten and left out when they finally become one

- Added two slots for security cadets
- Front security and maint access
- Basic security-related skills (to prevent them becoming MAA replacement)
- Minimal amount of skill points
- From the authority standpoint they have none, they can't arrest or do anything an MAA would do without MAA's supervision. All in all a crewman with basic sec access

Discussion thread on forums:
https://forums.baystation12.net/threads/security-cadet.6906/

Moonyman committed:
MAA Trainee Position
All other departments have trainees, security is no worse. Opened in loving memory of those poor crewmen who are willing to spend half of the round sitting in front of the XO's desk and by security to get their security trainee status only to get forgotten and left out when they finally become one

- Added two slots for security cadets
- Front security and maint access
- Basic security-related skills (to prevent them becoming MAA replacement)
- Minimal amount of skill points
- From the authority standpoint they have none, they can't arrest or do anything an MAA would do without MAA's supervision. A crewman with basic sec access

Moonyman committed:
BASIC skills

Nerev4r commented:
They'll still be armed just the same as MAAs, and dick antags just the same.

FreePrivateer commented:
@ParadoxSpace has a point. That and they should be trained to do the job before they get on the Charon. If we really need a trainee position, add MAA Trainee as an alt-title, and the sec department will just have to deal with training newbies.

Moonyman commented:
i don't see how they will be armed the same as MAAs without having MAA's access. And if you don't have the proper equipment trying to fight antags oftentimes leads to dying. MAA trainee as an alt title indulges this problem rather than solves it as alt titles have all the same access.
On another topic i don't know why "MAAs dicking antags" is posed as a bad thing, i was under impression that providing balance by opposing antags is the main purpose of security in game.

Moonyman committed:
crewman uniform instead of sec uniform

Moonyman commented:
wtf is a hex color check test?

MistakeNot4892 commented:
Hex values must be lowercase.

Moonyman committed:
lowercase hex

Minijar commented:
I believe this was a position before but was removed to stop players joining and just acting as regular MAAs and bypassing the MAA limit ingame. Is this any different?

CakeQ commented:
Yeah the issue is that even if they don't have the access the brig officer/COS will just give them equipment all the same.
We already have a big issue with security curbstomping mercs when at full capacity- We don't need more

Moonyman commented:
Why would CoS and/or BO give them weapons outside of dire situation?
And if the dire situation arises the entire crew is being armed (because half of them is military) and cadets don't really change the situation. The problem of mercs being destroyed is not solved by the lack of cadets but by moving away from military setting.

Acting like regular MAAs is controlled by station SOP. Cadets don't have the authority to do anything a regular MAA would do. If they try to arrest someone they can be taken in for illegal confinement as per SCG law. Not to mention they physically can't brig anyone as they lack access to the brig

According to my experience a non-armored non-armed crewman trying to rush a merc squad has an extremely short life expectancy and is likely to get bwoinked for that so i don't really see the issue

CakeQ commented:
If you honestly believe that security isn't going to give the trainees small e-guns just because they can then I've got some bad news for you.

Moonyman commented:
It's like saying "security will take lethals from the armory on green just because they can", yet they don't do that for some reason. A strawman argument based on lack of trust in the playerbase. I don't know a single BO or CoS that would neglect SOP just because they can, do you? If yes, point me in their direction please

Moonyman commented:
on a mode code-related note i have no idea why away site map got fucked up and how to fix it. I swear i'm not to blame

Minijar commented:
We're saying they'll do it because they do and have done in the past. It's precisely why this feature was removed in the past because people joined as trainees when the main MAA roles were full just to play as security and to get a gun.

Moonyman commented:
If they armed cadets without a good reason in the past why weren't such people dealt with OOCly and ICly?
p.s. As to they "do" it, i haven't seen a single instance of an MAA arming a crewman with guns for no reason
pss both arguments are anecdotal but meh

Nerev4r commented:
Moony, you're an Aurora player, so I'm sure you understand that cadets are /always/ given guns. We don't need to give them the opportunity.

afterthought2 commented on maps/torch/job/jobs.dm:
```diff
@@ -926,6 +926,43 @@
+		/datum/mil_rank/fleet/e2,
+	)
+
+	min_skill = list(	SKILL_BUREAUCRACY = SKILL_BASIC,
```
The design behind trainee position skill selections was that they start with relatively decent skill, but very few free points, making character creation simpler and less disaster-prone for new players. See how the other trainees do this.

I do like the idea of a sec trainee option, and find the balance concerns overblown. I never play sec in part due to having no good way to signal to the other sec players that I'm new to it.

Moonyman replied:
Yeah, i was consulting your trainee PR but in my opinion being a sec officer requires more common sense and situational awareness than skills (compare that with medical job where you literally mechanically can't do anything if you don't have relevant skills on trained and above). So i think basic skills +4 free points would do fine to get a hang of regs/procedures and at the same time discourage active combat and arrests (that seems to be a point of concern for a lot of people here for a couple of reasons)

afterthought2 replied:
Maybe I'm missing it, but you don't seem to set the free points to 4. You'd need to add
```
	skill_points = 4
	no_skill_buffs = TRUE
```
The latter is so that they don't get more/less points from things like age.

Moonyman commented:
Cadets on Aurora back when i was playing there were only given guns when every single officer was dead. At which point on bay you would bring in the military non-sec crew. So the balance is largely untouched

afterthought2 commented on maps/torch/job/jobs.dm:
```diff
@@ -926,6 +926,43 @@
+	department = "Security"
+	department_flag = SEC
+
+	total_positions = 2
```
Could set this to 1 to reduce balance issues. Given the overall higher pop due to hub changes, +1 sec player doesn't seem like the end of the world.

Moonyman committed:
Reducing the slot count to 1

Moonyman committed:
Fuck i forgot to set the skill points
- Set the skill point to 6 (two more than other trainees to somewhat compensate for lower skill levels)

Moonyman commented:
why does travis fuck up the away site tests every other run?

Higgin commented:
We don't need more security slots. Security should not have a training slot. 
As much security as we have right now and an easily-armed crew already puts the clamps on antags. 
Limiting cadets' skills doesn't matter as much as we might think; warm bodies matter most. We already have crewmen for warm bodies without overlapping into security.
The only adequate training for security comes from _actually playing security._
And whether up front or after they've been doing it for a bit, cadets _will_ treat the role like another security slot. 
If nobody else is on, you'll be expected to do arrests like corpsmen can do some surgeries; if guns get passed out, you'll be first in line and expected to fight. 

> If you honestly believe that security isn't going to give the trainees small e-guns just because they can then I've got some bad news for you.

> We're saying they'll do it because they do and have done in the past.

The bad news is: I would and we did.

FreePrivateer commented:
1) To clarify something here, despite it being slightly unrelated: The /entire/ crew is eligible to be armed in an emergency, not simply the military crew. Arming the crew should not favor fleet personnel over EC personnel, because they both have the same responsibilities. Its a roleplay style and background difference.

2) Assembling a crew levy takes time, access, and coordination (ignoring the fact that security /should/ normally be superior to the crew with their combat skills too.) Crew aren't getting immediately armed at blue alert, but there's a strong chance that your trainees are at least gonna get armor.

PRIMARILY) To be clear **every other department** has trainees as part of their own job slots. Corpsman Trainees and Engineer Trainees, positions that honestly require a lot more experience and training in the games mechanics, have to make due with trainees taking up engineering slots, and this is fine; It enables someone who isn't comfortable being a full engineer slowly acclimate to the role at their pace, because they still have access to all the skills, equipment, and areas a normal engineer has, but with the understanding that they're training.  Now that we have skill-based mechanics, playing a trainee with /no/ skills isn't as useful a thing because those skills are required to understand how the job mechanically works. Security should not be special,in this regard.

babydoll commented:
@Moonyman 

Please open a thread in the Gameplay Changes forum: https://forums.baystation12.net/view/gameplay-changes.104/

Make sure you include a link to the PR and the reasoning you want to open up with in the OP.

Moonyman commented:
Ideally every job should have a training slot most of all high impact slots like security and command. Learning about the job with high responsibilities without actually having to take those responsibilities would benefit everyone involved when trainee gets to finally pick the job. Not only that but security trainee position has a demand as i keep seing crewmen spending a lot of time trying to get into that position and failing due to command being busy.
As a security main i can with some amount of authority say that numbers are absolutely not the deciding factor in fights. Equipment and situational awareness matters much more. As a guy who dedicated a lot of time helping out cadets i can say that playing security is absolutely not the only way to learn how you should do it, an overwhelming majority of mistakes can be avoided by simply listening to a more experienced officer on a lecture.
Of course cadets will be the first one to be called when noone is aboard and security issue arises. It is a non-issue. There is already a practice of arming volunteers in emergencies in the absense of security which is the same exact thing.
Well as for bad news and the idea that they absolutely will be armed no matter what, i'll repeat myself: it's a strawman argument. Security does not arm themselves with lethals on greens nor will they arm cadets with guns on green. If you would personally do it then you're a bad CoS/BO and should probably avoid security roles. If you see someone else do it you can adress it both ICly and OOCly by talking to CO or admins and the problem will be quickly solved.

On the topic of mercs getting destroyed i'd say it's not exactly a favorable role in the first place. I mean you're attacking a ship staffed with SCG military, you'd think a direct approach is not something you would want to try with this setup. If mercs are meant to be a pew pew shooty shoot mode i can probably see to it being upped in firepower but that's a whole other story

PS i have a strong feeling that a failed Travis check will fix itself if rebooted. I can try to do a fake commit to prompt it unless there's an easier way to fix it.

afterthought2 commented:
While you can force-push an amended commit message to restart Travis (or just ask a dev), in this case it's probably best to wait for #22839 before restarting anyway.

Moonyman commented:
Link to a discussion thread added in the PR description

Moonyman committed:
Renamed security cadet to "master at arms trainee"

babydoll commented:
In light of community feedback and developer consensus, we have decided that we will not be merging this PR. Thank you for your contribution.
